File: data structures and algorithms/java/06_114.java
public static int recursiveFunc(int n, int depth, 
     Map<Integer, Integer> memo) {
  // Initialize memo if null
  if (memo == null) {
     memo = new HashMap<>();
  }
  
  // Create indentation based on recursion depth
  String indent = "  ".repeat(depth);
  System.out.println(indent + "Computing f(" + n + ")");
  
  // Return cached result if available
  if (memo.containsKey(n)) {
     System.out.println(indent + "Found in cache: " + 
        memo.get(n));
     return memo.get(n);
  }
  
  // Rest of function...
}

File: data structures and algorithms/cs/06_114.cs
using System;
using System.Collections.Generic;

public static int RecursiveFunc(int n, int depth, Dictionary<int, int> memo = null) {
  // Initialize memo if null
  if (memo == null) {
     memo = new Dictionary<int, int>();
  }
  
  // Create indentation based on recursion depth
  string indent = new string(' ', depth * 2);
  Console.WriteLine(indent + "Computing f(" + n + ")");
  
  // Return cached result if available
  if (memo.ContainsKey(n)) {
     Console.WriteLine(indent + "Found in cache: " + memo[n]);
     return memo[n];
  }
  
  // Rest of function...
}